relay/cache/cache_accessor.go

```diff
@@ -47,6 +48,9 @@ type cacheAccessor[K comparable, V any] struct {
 	// cache is the underlying cache that this wrapper manages.
 	cache Cache[K, V]
 
+	// weightCalculator is the function used to calculate the weight of a key-value pair.
+	calculator WeightCalculator[K, V]
```

Comment:
What is this calculator or the weight used for? 
I feel like abstractions like this make the code bit harder to read

Reply:
It's used to calculate the size of each individual item in the cache. Things like blobs are straight forward, you just take the length of the byte array. Chunks are much less straight forward, since they are stored in a more complex data structure.

I agree that these types of abstractions add complexity. I'm just not sure how to make the cache sufficiently generic as to work with both blobs and chunks without it.